### PR TITLE
 OCPBUGS-18089,OCPBUGS-18088: ovnkube: container scripts cleanup [v2]

### PR DESCRIPTION
**Pull request by dcbw**
Put common scripts into a bash script library and source that library from the ovnkube-node containers.

### CODE REVIEW

**Comment by dcbw** [comment]
/test e2e-aws-ovn-serial

`msg=Error: creating EC2 Instance: InvalidNetworkInterfaceID.NotFound: The networkInterface ID 'eni-034f7e54dd55aec50' does not exist`

**Comment by dcbw** [comment]
/test e2e-aws-ovn-local-to-shared-gateway-mode-migration

`msg=Error: creating ELBv2 network Load Balancer (ci-op-ys9ni2g9-416af-l9plq-int): SubnetNotFound: InvalidSubnetID.NotFound`

**Comment by dcbw** [comment]
/retest-required

hypershift failed before the network operator was even deployed in both cases

File: pkg/network/ovn_kubernetes.go [hunk 1]
```diff
@@ -486,7 +484,7 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 
 	if len(bootstrapResult.OVN.OVNKubernetesConfig.SmartNicModeNodes) > 0 {
 		data.Data["OVN_NODE_MODE"] = OVN_NODE_MODE_SMART_NIC
-		manifests, err = render.RenderTemplate(filepath.Join(manifestDir, manifestSubDir+"/ovnkube-node.yaml"), &data)
+		manifests, err = render.RenderTemplate(filepath.Join(manifestSubDir, "ovnkube-node.yaml"), &data)
```
Comment (jcaamano) [inline pkg/network/ovn_kubernetes.go:487]: nice

File: pkg/network/ovn_kubernetes.go [hunk 2]
```diff
@@ -416,10 +416,16 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	}
 	data.Data["ClusterManagerReplicas"] = clusterManagerReplicas
 
+	productFlavor := "self-hosted"
+	if bootstrapResult.OVN.OVNKubernetesConfig.HyperShiftConfig.Enabled {
+		productFlavor = "managed"
+	}
+	manifestSubDirBasePath := filepath.Join(manifestDir, "network/ovn-kubernetes", productFlavor)
```
Comment (jcaamano) [inline pkg/network/ovn_kubernetes.go:423]: this is a nit and don't need it on this PR

but isn't this supposed to be used as
```
manifestSubDirBasePath := filepath.Join(manifestDir, "network", "ovn-kubernetes", productFlavor)
```

Comment (jcaamano) [reply pkg/network/ovn_kubernetes.go:423]: I guess its fine, ignore me

Comment (dcbw) [reply pkg/network/ovn_kubernetes.go:423]: Yeah, it really should be; I chose not to rework that bit to keep the changes more minimal.

**Comment by jcaamano** [comment]
@dcbw we might have an issue with MTU migration. We relied on the fact that there were no rollouts for config file changes, where the MTU is configured. MTU changes were only incorporated after a rolling reboot triggered by MCO, but with this change the rollout due to MTU change can happen before the reboot. 

We might have to take the approach we took with SDN. Save the MTU configuration on a different file that won't cause a rollout on changes and then pass that file through CLI arg to ovnkube.

```
F1026 18:52:02.393045   36003 ovnkube.go:136] failed to start node network manager: failed to start default node network controller: interface MTU (1060) is too small for specified overlay MTU (1218)
```

I think we should improve the MTU check on OVN side to account for the routable MTU because there could be unexpected restarts as well. 

We can probably take either suggestion, although it would be good to have solve both counts. I can take the OVNK side of things.

Let me know what you think.

**Comment by dcbw** [comment]
`fatal: unable to access 'https://github.com/openshift/cluster-network-operator.git/': Could not resolve host: github.com`

/retest-required

**Comment by jluhrsen** [comment]
@dcbw, I know I want to back-port the commit in this PR to change [initialDelaySeconds to 10
](https://github.com/openshift/cluster-network-operator/pull/2060/commits/01c5f0462c4bdae984122e1b37b9287287ffea45)for 4.14. I can just make a small PR to do that, but wondering if you thought any of the other
20 commits are important enough to bring in with it. I just tried a /cherry-pick but it had some
merge conflict. I'll see if that's easy to resolve locally and make a new PR with all these changes,
but wanted your opinion either way please.

**Comment by jluhrsen** [comment]
> @dcbw, I know I want to back-port the commit in this PR to change [initialDelaySeconds to 10 ](https://github.com/openshift/cluster-network-operator/pull/2060/commits/01c5f0462c4bdae984122e1b37b9287287ffea45)for 4.14. I can just make a small PR to do that, but wondering if you thought any of the other 20 commits are important enough to bring in with it. I just tried a /cherry-pick but it had some merge conflict. I'll see if that's easy to resolve locally and make a new PR with all these changes, but wanted your opinion either way please.

Ah, I see you already have something in 4.14:
https://github.com/openshift/cluster-network-operator/pull/2090

**Pull request merged**